### ISSUE TITLE
Hyperparameters Study

### ISSUE THREAD

**Issue by rezaBarzgar** [fani-lab/OpeNTF#236]
### This issue reports the results of my SuperLoss hyperparameter study on IMDB and DBLP.

You can find results in detail [here](https://docs.google.com/spreadsheets/d/1oatYd2uMXOah8RQJsvaT2NIwxhh6CfssaneVLuYczQY/edit?usp=sharing), in sheets **DBLP hyperparameter SUPERLOSS** and **IMDB hyperparameter SUPERLOSS**.

### SuperLoss has two hyperparameters which are **&tau;** and **&lambda;**:

- **&tau;**: A threshold that separates easy experts from difficult ones based on their respective losses.
- **&lambda;**: Regularization hyperparameter.

In this hyperparameter study, 5 different permutations of **&tau;** and **&lambda;** were tested on 2 different datasets (IMDB and DBLP):

- &tau; 0.7 - &lambda; 0.0 (finished)
- &tau; 0.7 - &lambda; 0.9 (finished)
- &tau; 0.0 - &lambda; 0.0 (finished)
- &tau; 0.5 - &lambda; 0.5 (finished)
- &tau; 0.0 - &lambda; 0.9 (finished) (Original values for the results that we reported in our papers)
- &tau; 0.9 - &lambda; 0.9 (running)
- &tau; 0.9 - &lambda; 0.0 (finished)

### Analysis of the results:

- This hyperparameter study **supports** our claim that loss-based curriculum learning techniques are ineffective for Bayesian neural networks.
- Based on the results, it can be concluded that &lambda; has a direct effect on the performance of models on team formation task while &tau; has the reverse effect. A higher value of &lambda; is better, a lower value of &tau; is better. The best results are gained from &tau; 0.0 - &lambda; 0.9.

**Comment by rezaBarzgar** [comment]
@hosseinfani 

Up until now, I've conducted the following experiments that we can include in our contributions for our next paper:

- Created a static curriculum based on popularity labels. The results are aligned with our claim that dynamic CL is a better fit for the Team Recommendation task.
- Conducted a hyperparameter study on superloss to examine the effects of different hyperparameters on the Team Recommendation task. It's worth mentioning that parametric CL (data parameters) has no additional hyperparameters to study.

I would appreciate it if you could review the results sheet and share your feedback.

Note: The results for the static curriculum will be updated on the sheet.

**Comment by hosseinfani** [comment]
this has been done in cl-wsdm25 branch and reported in our accepted paper but needs integration with main line.
https://hosseinfani.github.io/res/papers/2025_WSDM_Adaptive_Loss-based_Curricula_for_Neural_Team_Recommendation.pdf
https://doi.org/10.1145/3701551.3703574
